### ISSUE TITLE
Migrations won't be applied because the table is not found due to capital letters usage

### ISSUE THREAD

**Issue by ctrlmaniac** [prisma/prisma#21367]
### Bug description

When the table is created in the migration.sql the table name is capitalized:

```sql
-- CreateTable
CREATE TABLE `Ticket`
```

But when I alter the table, the migration.sql uses a lowercase table name

```sql
ALTER TABLE `ticket` DROP COLUMN `tipoManutezione`
```

I have to manually edit the migration.sql file but when I do it the database will be reset by prisma.

To sync the database I have to run the following command `npx prisma db push`

### Expected behavior

```-- CreateTable
CREATE TABLE `Ticket`
```
and

```sql
ALTER TABLE `Ticket` DROP COLUMN `tipoManutezione`
```

### Prisma information

```prisma
enum TipoManutenzione {
  RICHIESTA
  ORDINE
}

model Ticket {
  id                  Int              @id @default(autoincrement())
  tipoManutenzione    TipoManutenzione @default(RICHIESTA)
}
```

### Environment & setup

- OS: Ubuntu 22.04 LTS
- Database: MySql
- Node.js version: v18.18.0

### Prisma Version

```
prisma                  : 5.4.1
@prisma/client          : 5.4.1
Current platform        : debian-openssl-3.0.x
Query Engine (Node-API) : libquery-engine 2f302df92bd8945e20ad4595a73def5b96afa54f (at node_modules/@prisma/engines/libquery_engine-debian-openssl-3.0.x.so.node)
Schema Engine           : schema-engine-cli 2f302df92bd8945e20ad4595a73def5b96afa54f (at node_modules/@prisma/engines/schema-engine-debian-openssl-3.0.x)
Schema Wasm             : @prisma/prisma-schema-wasm 5.3.1-2.61e140623197a131c2a6189271ffee05a7aa9a59
Default Engines Hash    : 2f302df92bd8945e20ad4595a73def5b96afa54f
Studio                  : 0.494.0
```

**Comment by janpio** [comment]
When you create the table via migrations, how is it actually called in the database, `Ticket` or `ticket`?

**Comment by ctrlmaniac** [comment]
In my local mysql database the table is called `ticket` but in production the database is called `Ticket` so in production the migration won't be applied!

**Comment by janpio** [comment]
Ok, that is a problem - both your databases need to be configured the same way to execute a `CREATE TABLE` with the same end result. Especially if you might create migrations on both systems, Prisma needs a stable basis to act on.

**Comment by ctrlmaniac** [comment]
OOT: do you know how to do this?

**Comment by janpio** [comment]
Usually it is a configuration option like https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_lower_case_table_names or similar. You might find older issues here about this as well when you start searching with some relevant terms. You are definitely not the first person to have this challenge with MySQL.

**Comment by ctrlmaniac** [comment]
However this won't be necessery if there were consinstency in how the migration file is built, isn't it?

**Comment by janpio** [comment]
I am not sure what you are suggesting. Prisma always uses the name defined in `schema.prisma` for migration files.

**Comment by ctrlmaniac** [comment]
Sometimes it doesn't respect the case

**Comment by janpio** [comment]
I don't think that is actually possible. That would require Prisma to apply a "lowercase" function to the model name. We don't do that. 

Do you have a reproduction for getting the lowercase `ticket` into the migration SQL?

**Comment by Azasratruth** [comment]
Getting the same error on windows. I thought it was just because windows 11 is case insensitive but apparently not.

Prisma creates tables according to the schema in production but locally it creates the tables with lowercase. 

And it seems like, for the migrations with alter table, prisma is fetching table names from the db and not the schema file. 

@ctrlmaniac, were you able to solve it?

**Comment by DiogoS** [comment]
Still happening with the last version.
